src/main/java/it/gov/pagopa/payhub/activities/activity/treasury/TreasuryOpiIngestionActivityImpl.java

```diff
@@ -26,50 +28,87 @@
 @Lazy
 @Component
 public class TreasuryOpiIngestionActivityImpl implements TreasuryOpiIngestionActivity {
-  private final IngestionFlowFileType ingestionflowFileType;
-  private final IngestionFlowFileDao ingestionFlowFileDao;
-  private final IngestionFlowFileRetrieverService ingestionFlowFileRetrieverService;
+    private final IngestionFlowFileType ingestionflowFileType;
+    private final IngestionFlowFileDao ingestionFlowFileDao;
+    private final IngestionFlowFileRetrieverService ingestionFlowFileRetrieverService;
+    private final TreasuryUnmarshallerService treasuryUnmarshallerService;
+
+
+    public TreasuryOpiIngestionActivityImpl(
+                                            IngestionFlowFileDao ingestionFlowFileDao,
+                                            IngestionFlowFileRetrieverService ingestionFlowFileRetrieverService,
+                                            TreasuryUnmarshallerService treasuryUnmarshallerService) {
+        this.ingestionflowFileType = IngestionFlowFileType.OPI;
+        this.ingestionFlowFileDao = ingestionFlowFileDao;
+        this.ingestionFlowFileRetrieverService = ingestionFlowFileRetrieverService;
+        this.treasuryUnmarshallerService = treasuryUnmarshallerService;
+    }
 
 
-  public TreasuryOpiIngestionActivityImpl(IngestionFlowFileDao ingestionFlowFileDao,
-                                          IngestionFlowFileRetrieverService ingestionFlowFileRetrieverService) {
-    this.ingestionflowFileType = IngestionFlowFileType.OPI;
-    this.ingestionFlowFileDao = ingestionFlowFileDao;
-    this.ingestionFlowFileRetrieverService = ingestionFlowFileRetrieverService;
-  }
+    @Override
+    public TreasuryIufResult processFile(Long ingestionFlowFileId) {
+        List<String> iufIuvList = new ArrayList<>();
+        List<Path> ingestionFlowFiles = null;
+        AtomicBoolean success = new AtomicBoolean(true);
+        try {
+            IngestionFlowFileDTO ingestionFlowFileDTO = findIngestionFlowFileRecord(ingestionFlowFileId);
 
+            ingestionFlowFiles = retrieveFiles(ingestionFlowFileDTO);
 
-  @Override
-  public TreasuryIufResult processFile(Long ingestionFlowFileId) {
-    List<String> iufIuvList = new ArrayList<>();
-    try {
-      IngestionFlowFileDTO ingestionFlowFileDTO = findIngestionFlowFileRecord(ingestionFlowFileId);
+        } catch (Exception e) {
+            log.error("Error during TreasuryOpiIngestionActivity ingestionFlowFileId {}", ingestionFlowFileId, e);
+            return new TreasuryIufResult(Collections.emptyList(), false);
+        }
 
-      List<Path> ingestionFlowFiles = retrieveFiles(ingestionFlowFileDTO);
+        if (ingestionFlowFiles != null && !ingestionFlowFiles.isEmpty()) {
```

Comment:
ingestionFlowFileRetrieverService could return null?

Reply:
it was an additional control for test. Condition removed